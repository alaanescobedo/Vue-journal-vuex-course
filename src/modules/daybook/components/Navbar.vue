<template>
  <nav class="navbar bg-primary">
    <a @click="goToHome" class="navbar-brand text-white pointer">
      <img
        src="@/assets/logo.png"
        alt="logo"
        class="d-inline-block align-text-top mx-2"
        height="24"
      />
      Daybook
    </a>
    <div class="d-flex">
      <button class="btn btn-outline-info mx-2"><i class="fa fa-sign-out-alt"></i></button>
    </div>
  </nav>
</template>

<script>
export default {
  methods: {
    goToHome() {
      this.$router.push({ name: 'Home' });
    },
  },
};
</script>
