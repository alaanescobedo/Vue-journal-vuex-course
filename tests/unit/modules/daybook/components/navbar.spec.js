import { shallowMount } from '@vue/test-utils';
import Navbar from '@/modules/daybook/components/Navbar';

import createVuexStore from '../../../mock/mock-store';
import { testUser } from '../../../mock/test-auth-state';

const mockRouter = {
  push: jest.fn(),
};

jest.mock('vue-router', () => ({
  useRouter: () => mockRouter,
}));

describe('Navbar Component', () => {
  const store = createVuexStore(testUser('authenticated'));

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('debe mostrar el componente correctamente', () => {
    const wrapper = shallowMount(Navbar, {
      global: {
        plugins: [store],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
  test('click en el logout, debe cerrar sesion y redireccionar', async () => {
    const wrapper = shallowMount(Navbar, {
      global: {
        plugins: [store],
      },
    });

    await wrapper.find('button').trigger('click');

    expect(mockRouter.push).toHaveBeenCalledTimes(1);
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'login' });
    expect(store.state.auth).toEqual({
      user: null,
      status: 'not-authenticated',
      idToken: null,
      refreshToken: null,
    });
  });
});
