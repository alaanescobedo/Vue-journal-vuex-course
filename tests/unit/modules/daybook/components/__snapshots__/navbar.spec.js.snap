// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component debe mostrar el componente correctamente 1`] = `
<nav class="navbar bg-primary"><a class="navbar-brand text-white pointer"><img src="" alt="logo" class="d-inline-block align-text-top mx-2" height="24"> Daybook </a>
  <h4 class="text-white me-auto my-auto">Hola Test Existing User!</h4>
  <div class="d-flex"><button class="btn btn-outline-info mx-2"><i class="fa fa-sign-out-alt"></i></button></div>
</nav>
`;
